env/dev/secretsmanager.tf

```diff
@@ -0,0 +1,166 @@
+# create the secretsmanager secret
+resource "aws_secretsmanager_secret" "sm_secret" {
+  name = "${var.app}-${var.environment}"
+  tags = "${var.tags}"
+  policy = "${data.aws_iam_policy_document.sm_resource_policy_doc.json}"
+}
+
+# get the saml user info so we can get the unique_id
+data "aws_iam_role" "saml_role" {
+  name = "${var.saml_role}"
+}
+
+# resource policy doc that limits access to secret
+data "aws_iam_policy_document" "sm_resource_policy_doc" {
+  statement = {
+    sid = "DenyWriteToAllExceptSAMLUsers"
+    effect = "Deny"
+    principals = {
+      type = "AWS"
+      identifiers = ["*"]
+    }
+    actions = [
+      "secretsmanager:CancelRotateSecret",
+      "secretsmanager:CreateSecret",
+      "secretsmanager:DeleteSecret",
+      "secretsmanager:PutSecretValue",
+      "secretsmanager:RestoreSecret",
+      "secretsmanager:RotateSecret",
+      "secretsmanager:TagResource",
+      "secretsmanager:UntagResource",
+      "secretsmanager:UpdateSecret",
+      "secretsmanager:UpdateSecretVersionStage",
+    ]
+    resources = [
+      "arn:aws:secretsmanager:${var.region}:${data.aws_caller_identity.current.account_id}:secret:${var.app}-${var.environment}-*"
+    ]
```

Comment:
Can you use `${aws_secretsmanager_secret.sm_secret.arn}` here instead which would lock it down to the exact secret?

Reply:
We could, but then future secrets for that env would be limited.  My concern is the case where someone needs to separate their secrets (for whatever reason).

Reply:
How would I make a "future secret for that env"?  Wouldn't the service complain about a new service with the same name?